onnxruntime/contrib_ops/cpu/skip_layer_norm_helper.h

```diff
@@ -33,49 +29,150 @@ Status CheckInputs(const T* input,
                            "last two dimensions of skip needs to be same as input");
   }
 
+  return Status::OK();
+}
+
+template <typename T>
+Status CheckGamma(const T* gamma, int hidden_size_check) {
   const auto& gamma_dims = gamma->Shape().GetDims();
+
   if (gamma_dims.size() != 1) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                            "gamma is expected to have 1 dimension, got ", gamma_dims.size());
   }
+
   if (gamma_dims[0] != hidden_size_check) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                            "Last dimension of gamma and input does not match");
   }
 
+  return Status::OK();
+}
+
+template <typename T>
+Status CheckBeta(const T* beta, int hidden_size_check) {
   if (nullptr != beta) {
     const auto& beta_dims = beta->Shape().GetDims();
+
     if (beta_dims.size() != 1) {
       return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                              "beta is expected to have 1 dimension, got ", beta_dims.size());
     }
+
     if (beta_dims[0] != hidden_size_check) {
       return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                              "Last dimension of beta and input does not match");
     }
   }
 
+  return Status::OK();
+}
+
+template <typename T>
+Status CheckBias(const T* bias, int hidden_size_check) {
   if (nullptr != bias) {
     const auto& bias_dims = bias->Shape().GetDims();
+
     if (bias_dims.size() != 1) {
       return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                              "bias is expected to have 1 dimension, got ", bias_dims.size());
     }
+
     if (bias_dims[0] != hidden_size_check) {
       return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                              "Last dimension of bias and input does not match");
     }
   }
+
+  return Status::OK();
+}
+
+}  // anonymous namespace
+
+template <typename T>
+Status CheckInputs(const T* input,
+                   const T* skip,
+                   const T* gamma,
+                   const T* beta,
+                   const T* bias,
+                   int hidden_size_check,
+                   size_t input_dims_size_check) {
+  if (input_dims_size_check != 3 && input_dims_size_check != 2) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "input is expected to have 3 or 2 dimensions, got ", input_dims_size_check);
+  }
+
+  auto status = CheckSkip<T>(input, skip, input_dims_size_check);
+  if (status != Status::OK()) {
+    return status;
+  }
+
+  status = CheckGamma<T>(gamma, hidden_size_check);
+  if (status != Status::OK()) {
+    return status;
+  }
+
+  status = CheckBeta<T>(beta, hidden_size_check);
+  if (status != Status::OK()) {
+    return status;
+  }
+
+  status = CheckBias<T>(bias, hidden_size_check);
+  if (status != Status::OK()) {
+    return status;
+  }
+
+  return Status::OK();
+}
+
+template <typename T>
+Status CheckPotentiallyPrepackedInputs(const T* input,
```

Comment:
Any reason to use template function as T is always Tensor?

Reply:
I wanted to be consistent with `CheckInputs` (which was this way before I updated the code) without doing unnecessary refactoring